core/src/main/java/org/mule/runtime/core/internal/construct/AbstractPipeline.java

```diff
@@ -440,8 +438,22 @@ protected void doDispose() {
     super.doDispose();
   }
 
+  /**
+   * Determines is blocking synchronous code path should be used. This is used in the following cases:
+   * <ol>
+   * <li>i) If a transaction is active and a processing strategy supporting transactions is configured. (synchronous or default
+   * strategies)</li>
+   * <li>i) If {@link LegacySynchronousProcessingStrategyFactory} is configured as the processing strategy.</li>
+   * </ol>
+   * 
+   * @return true if blocking synchronous code path should be used, false otherwise.
+   */
   protected boolean useBlockingCodePath() {
-    return isTransactionActive() || processingStrategy.isSynchronous();
+    return (isTransactionActive() &&
+        (processingStrategyFactory instanceof DefaultFlowProcessingStrategyFactory
```

Comment:
`processingStrategyFactory instanceof DefaultFlowProcessingStrategyFactory` condition is repeated